### PR TITLE
Add an example demoing rewrite-clj.

### PR DESCRIPTION
**Pull request by jwhitlark**
Please answer the following questions and leave the below in as part of your PR.

- [X] I have read the [developer documentation](https://github.com/BetterThanTomorrow/joyride/blob/master/CONTRIBUTE.md).

No, this is an example.
- [X] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/babashka/blob/master/doc/dev.md#start-with-an-issue-before-writing-code).

Not a code change
- [X] This PR contains a [test](https://github.com/BetterThanTomorrow/joyride/tree/master/vscode-test-runner/workspace-1/.joyride/src/integration_test) to prevent against future regressions

Not a code change
- [X] I have updated the **\[Unreleased\]** section of the [CHANGELOG.md](https://github.com/BetterThanTomorrow/joyride/blob/master/CHANGELOG.md) file with a link to the addressed issue.

### CODE REVIEW

File: examples/.joyride/src/port_arrow_form.cljs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+(defn main []
+  (log "port-arrow-form started")
+  (-> (p/let [editor ^js vscode/window.activeTextEditor
+              selection (e/current-selection)
+              source (str "[ " (e/current-selection-text) " ]")
+              zloc (z/of-string source)
+              [lhs arrow rhs] (get-arrow-clause (z/down zloc))]
+        (e/delete-range! editor selection)
+        (e/insert-text!+ (str `(~'is (~'= ~rhs ~lhs)))))))
```
Comment (PEZ) [inline examples/.joyride/src/port_arrow_form.cljs:30]: As discussed in Slack. Maybe it is time for `e/replace-text!+`.

Comment (PEZ) [reply examples/.joyride/src/port_arrow_form.cljs:30]: Besides, I think `e/delete-range!` lacks the `+` in its name (which we use for promise-returning functions). If you feel up to it, you are welcome to fix that too in this PR.

**Comment by jwhitlark** [comment]
@PEZ Please let me know if there's any more changes needed.

File: examples/.joyride/src/util/editor.cljs [hunk 2]
```diff
@@ -35,6 +35,22 @@
       (p/catch (fn [e]
                  (js/console.error e)))))
 
+;; update name, but don't break existing scripts
+(def delete-range!+ delete-range!)
+
+(defn replace-range!+
+  "Defaults to the current selection."
+  ([^js text]
+   (replace-range! text vscode/window.activeTextEditor (.-active (current-selection)) (current-selection)))
```
Comment (PEZ) [inline examples/.joyride/src/util/editor.cljs]: Your REPL probably knows what `replace-range!` is, but it's not defined in the file, is it? 😄

File: examples/.joyride/src/util/editor.cljs [hunk 3]
```diff
@@ -35,6 +35,22 @@
       (p/catch (fn [e]
                  (js/console.error e)))))
 
+;; update name, but don't break existing scripts
+(def delete-range!+ delete-range!)
```
[the remark this answers is not in the document]
Comment (jwhitlark) [reply examples/.joyride/src/util/editor.cljs]: How's that?

File: examples/.joyride/src/util/editor.cljs (hunk 2 above)
Comment (jwhitlark) [reply examples/.joyride/src/util/editor.cljs]: oops.  😳

**Comment by jwhitlark** [comment]
Should all be fixed.

**Comment by PEZ** [comment]
Awesome. I think it is worth a mention in examples/README.md as well. 😄

File: examples/README.md [hunk 4]
```diff
@@ -88,6 +88,30 @@ the default Windows comment keyboard shortcut.
 }
 ```
 
+## Structural Editing via rewrite-clj
+
+`.joyride/src/port_arrow_form.cljs`
```
[the remark this answers is not in the document]
Comment (jwhitlark) [reply examples/README.md]: I moved it to scripts.  Let me know if that's the right choice.  I haven't looked up how to set up a proper dev env for this stuff, so I'm kinda guessing.

[the remark this answers is not in the document]
Comment (PEZ) [reply examples/README.md]: That's the right place for it. Thanks!

**Pull request merged**